pyfakefs/deprecator.py

```diff
@@ -0,0 +1,69 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utilities for handling deprecated functions."""
+
+import functools
+import warnings
+
+
+class Deprecator(object):
+    """Decorator class for adding deprecated functions.
+   
+    Warnings are switched off by default.
+    To enable deprecation warnings, use:
+
+    >>>  from pyfakefs.deprecator import Deprecator
+    >>>
+    >>>  Deprecator.show_warning = True
+    """
+
+    show_warning = False
```

Comment:
I think `show_warnings` would be a better name.

Reply:
Thanks, I agree - done.